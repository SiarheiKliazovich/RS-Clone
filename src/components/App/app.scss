@import "../../sass/mixins";
@import "../../sass/normalize.scss";
@import "../../sass/variables";
@import "../../sass/general";
