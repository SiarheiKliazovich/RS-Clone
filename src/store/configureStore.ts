/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
