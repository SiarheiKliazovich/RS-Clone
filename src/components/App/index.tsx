import "./app.scss";
import { Routes, Route, BrowserRouter } from "react-router-dom";
import { FunctionComponent } from "react";
import { useSelector } from "react-redux";
import Dashboard from "../Dashboard";
import SignUp from "../SignUp";
import SignIn from "../SignIn";
import Home from "../Home";
import TieMarket from "../TieMarket";
import { IStore } from "../../interfaces/store";
import classNames from "classnames";
import Configurator from "../Configurator";
import MyProfile from "../MyProfile";

const App: FunctionComponent = () => {
  const { isNightMode } = useSelector((state: IStore) => state.appInterface);
  document.body.className = classNames("", { "body--night-mode": isNightMode });

  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/configurator" element={<Configurator />} />
        <Route path="/tiemarket" element={<TieMarket />} />
        <Route path="/dashboard/*" element={<Dashboard />} />
        <Route path="/sign-in" element={<SignIn />} />
        <Route path="/sign-up" element={<SignUp />} />
        <Route path="/profile" element={<MyProfile />} />
        {/* <Route path="/my-ties" element={<MyTies />} />
        <Route path="/my-orders" element={<MyOrders />} /> */}
      </Routes>
    </BrowserRouter>
  );
};

export default App;
