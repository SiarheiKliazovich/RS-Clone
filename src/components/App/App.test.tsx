import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import App from ".";

const mockStore = configureStore([]);

describe("App", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      auth: {
        user: {
          role: "USER",
        },
      },
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
      langReducer: {
        lang: "en",
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
