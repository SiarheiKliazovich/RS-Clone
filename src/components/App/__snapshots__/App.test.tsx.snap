// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<header
  class="header"
  style="background-color: rgb(0, 0, 0);"
>
  <div
    class="container"
  >
    <div
      class="header__wrapper"
    >
      <a
        class="header__link"
        href="/"
      >
        <span
          class="header__logo"
        />
      </a>
      <div
        class="nav__hamburger_wrapper"
      >
        <span
          class="nav__hamburger"
        />
      </div>
      <div
        class="nav-sidebar"
        style="background-color: rgb(0, 0, 0);"
      >
        <div
          class="nav-sidebar-wrapper"
        >
          <ul
            class="list-nav-sidebar"
          >
            <li
              class="nav-sidebar-item"
            >
              <span
                class="active-link"
              >
                Home
              </span>
            </li>
            <li
              class="nav-sidebar-item"
            >
              <a
                class="nav-sidebar-link"
                href="/configurator"
              >
                Configurator
              </a>
            </li>
            <li
              class="nav-sidebar-item"
            >
              <a
                class="nav-sidebar-link"
                href="/tie-market"
              >
                Tie Market
              </a>
            </li>
          </ul>
        </div>
      </div>
      <ul
        class="header__list"
      >
        <li
          class="header__item"
        >
          <div
            class="language"
          >
            <button
              class="button__lang"
              title="button__lang"
            >
              <img
                alt="Current language"
                class="language__icon"
                src="flag-us.jpg"
              />
            </button>
            <span
              class="language__title"
            >
              en
            </span>
          </div>
        </li>
        <li
          class="header__item"
          title="header__profile"
        >
          <span
            class="header__profile"
          />
        </li>
      </ul>
    </div>
  </div>
</header>
`;
